src/data/fintype/basic.lean

```diff
@@ -902,6 +902,24 @@ begin
     exact subtype.eq (H x.2 y.2) }
 end
 
+lemma finset.card_le_one_iff_subset_singleton [nonempty α] {s : finset α} :
+  s.card ≤ 1 ↔ ∃ (x : α), s ⊆ {x} :=
+begin
+  split,
+  { assume H,
+    by_cases h : ∃ x, x ∈ s,
+    { rcases h with ⟨x, hx⟩,
+      refine ⟨x, λ y hy, _⟩,
+      rw finset.card_le_one_iff.1 H hy hx,
+      simp only [mem_singleton] },
+    { push_neg at h,
+      inhabit α,
+      exact ⟨default α, λ y hy, (h y hy).elim⟩ } },
+  { rintros ⟨x, hx⟩,
+    rw ← card_singleton x,
+    exact card_le_of_subset hx }
```

Comment:
I think this (and the one before it) ought to go in `data/finset/basic`, next to the existing `finset.card_le_one`: `finset.card_le_one_iff` is a trivial modification of that (even if the proof doesn't appear so).

Reply:
You're absolutely right. I have moved it to the right place, together with the other lemmas on finsets that were in `fintype.lean`.